### PR TITLE
add git-csa

### PR DESCRIPTION
**Pull request by toshipp**
git-csa is a git subcommand that is the same as git-neco but does not use zenhub.
It uses native github feature that links PRs to issues by the ID.

issue: cybozu/csa#42

### CODE REVIEW

File: git-csa/cmd/git.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
+package cmd
```
Comment (llamerada-jp) [inline git-csa/cmd/git.go:1]: According to [Standard Go Project Layout](https://github.com/golang-standards/project-layout), I think it is better to move git.go and github.go to /internal or /pkg.

File: git-csa/cmd/github.go [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+//    - git@github.com:<owner>/<name>.git -> (owner, name)
+// 3. Other
+//    - <owner>/<name> -> (owner, name)
+func ExtractGitHubRepositoryName(repo string) (string, string, error) {
+	// For SCP-like address
+	// e.g. git@github.com:user/repo.git -> user/repo.git
+	reg, err := regexp.Compile(`^([a-zA-Z0-9_]+)@([a-zA-Z0-9._-]+):(.*)$`)
+	if err != nil {
+		return "", "", err
+	}
```
Comment (llamerada-jp) [inline git-csa/cmd/github.go:67]:
```suggestion
	reg := regexp.MustCompile(`^([a-zA-Z0-9_]+)@([a-zA-Z0-9._-]+):(.*)$`)
```

File: git-csa/cmd/github.go [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+	ID        string
+	Number    int
+	Permalink string
+}
+
+// ExtractGitHubRepositoryName extracts repository owner and name from a string.
+// This function treats following syntax.
+// 1. GitHub URL
+//    - https://github.com/<owner>/<name>.git -> (owner, name)
+// 2. SCP-like address
```
Comment (llamerada-jp) [inline git-csa/cmd/github.go:57]: It looks like this function only supports pattern 2.

File: git-csa/cmd/draft.go [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+		if !strings.HasSuffix(firstBody, "\n") {
+			firstBody += "\n"
+		}
+		firstBody += issueLink
+	}
+
+	_, err = createPR(ctx, gc, curRepo, defBranch, branch, title, firstBody, draft)
+	if err != nil {
+		return err
+	}
```
Comment (toshipp) [inline git-csa/cmd/draft.go:149]: I write only this section, another part is copied from `git-neco`

**Comment by toshipp** [comment]
I found a bug that adds miss link when no issue id is specified.

**Review by satoru-takeuchi — APPROVED** [review]

File: git-csa/cmd/draft.go [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+		title = firstSummary
+	}
+
+	issueLink := fmt.Sprintf("issue: %s/%s#%d", issueRepo.Owner, issueRepo.Name, draftOpts.issue)
+	if !strings.Contains(firstBody, issueLink) {
+		if !strings.HasSuffix(firstBody, "\n") {
+			firstBody += "\n"
+		}
+		firstBody += issueLink
+	}
```
Comment (llamerada-jp) [inline git-csa/cmd/draft.go]:
```suggestion
	signedOff := "Signed-off-by:"
	exp := regexp.MustCompile(signedOff)

	if !strings.Contains(firstBody, issueLink) {
		if !strings.HasSuffix(firstBody, "\n") {
			firstBody += "\n"
		}
		
		isFirst := true
		if exp.Match([]byte(firstBody)) {
			firstBody = exp.ReplaceAllStringFunc(firstBody, func(src string) string {
				if isFirst {
					isFirst = false
					return issueLink + "\n" + src
				}
				return src
			})
		} else {
			firstBody += issueLink
		}
	}
```

**Review by llamerada-jp — APPROVED** [review]
LGTM

**Pull request merged**